x/merkledb/trieview.go

```diff
@@ -246,10 +250,23 @@ func (t *trieView) calculateNodeIDs(ctx context.Context) error {
 			}
 		}
 
+		// If the root has no value and only a single child, that child can become the root.
+		// Use valueDigest instead of value because proof verification only sets the digest
```

Comment:
If the root has no value and a single child, why is it `t.root` here? Shouldn't its child become the root in `remove`/`insert`?

Reply:
There is a tricky case where it was hard to detect in insert/remove.  It was easier to catch it here.